Integrating StarRocks with Metabase

**TL;DR**: it seems Metabase MariaDB/ MySQL driver tries to call `time_format(TIME, varchar)` as part of the `sync-timezone` routine, but StarRocks doesn't provide such function.

### Steps to reproduce the behavior (Required)
1. Start a Metabase instance (either locally or SaaS).
2. Connect Metabase with StarRocks by following https://docs.starrocks.io/docs/integrations/BI_integrations/Metabase/.
3. Trigger a database sync.

### Expected behavior (Required)
1. See StarRocks tables on Metabase.

### Real behavior (Required)
1. Metabase succeeds on validating the connection.
2. I cannot see any tables as part of the database connection I've just created.
3. By checking Metabase logs, I can see a failure while calling `sync-timezone` routine (it seems to be the cause, but not sure).

### StarRocks version (Required)
3.2.3-a40e2f8
 
### Metabase version
Metabase Cloud (v1.48.7)/ v0.48.7

### Metabase logs
```
[6332a73e-e2d8-4264-ab1f-0bd8a95a0d3f] 2024-02-28T07:39:00-03:00 INFO metabase.task.sync-databases Starting sync task for Database 35.
[6332a73e-e2d8-4264-ab1f-0bd8a95a0d3f] 2024-02-28T07:39:00-03:00 WARN metabase.driver.mysql 

********************************************************************************
WARNING: Metabase only officially supports MySQL 5.7/MariaDB 10.2 and above.
All Metabase features may not work properly when using an unsupported version.
********************************************************************************

[6332a73e-e2d8-4264-ab1f-0bd8a95a0d3f] 2024-02-28T07:39:00-03:00 INFO metabase.sync.util STARTING: Sync metadata for mysql Database 35 'StarRocks'
[6332a73e-e2d8-4264-ab1f-0bd8a95a0d3f] 2024-02-28T07:39:00-03:00 INFO metabase.sync.util STARTING: step 'sync-dbms-version' for mysql Database 35 'StarRocks'
[6332a73e-e2d8-4264-ab1f-0bd8a95a0d3f] 2024-02-28T07:39:00-03:00 INFO metabase.sync.util FINISHED: step 'sync-dbms-version' for mysql Database 35 'StarRocks' (151.5 ms)
[6332a73e-e2d8-4264-ab1f-0bd8a95a0d3f] 2024-02-28T07:39:00-03:00 INFO metabase.sync.util STARTING: step 'sync-timezone' for mysql Database 35 'StarRocks'
[6332a73e-e2d8-4264-ab1f-0bd8a95a0d3f] 2024-02-28T07:39:01-03:00 WARN metabase.sync.util Error running step 'sync-timezone' for mysql Database 35 'StarRocks'
java.sql.SQLTransientConnectionException: (conn=601) Getting analyzing error from line 1, column 73 to line 1, column 173. Detail message: No matching function with signature: time_format(TIME, varchar).
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.createException(ExceptionFactory.java:79)
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.create(ExceptionFactory.java:158)
	at org.mariadb.jdbc.MariaDbStatement.executeExceptionEpilogue(MariaDbStatement.java:262)
	at org.mariadb.jdbc.ClientSidePreparedStatement.executeInternal(ClientSidePreparedStatement.java:229)
	at org.mariadb.jdbc.ClientSidePreparedStatement.execute(ClientSidePreparedStatement.java:149)
	at org.mariadb.jdbc.ClientSidePreparedStatement.executeQuery(ClientSidePreparedStatement.java:163)
	at com.mchange.v2.c3p0.impl.NewProxyPreparedStatement.executeQuery(NewProxyPreparedStatement.java:1471)
	at clojure.java.jdbc$execute_query_with_params.invokeStatic(jdbc.clj:1090)
	at clojure.java.jdbc$execute_query_with_params.invoke(jdbc.clj:1084)
	at clojure.java.jdbc$db_query_with_resultset_STAR_.invokeStatic(jdbc.clj:1113)
	at clojure.java.jdbc$db_query_with_resultset_STAR_.invoke(jdbc.clj:1093)
	at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1182)
	at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
	at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1160)
	at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
	at metabase.driver.mysql$fn__106633.invokeStatic(mysql.clj:205)
	at metabase.driver.mysql$fn__106633.invoke(mysql.clj:195)
	at clojure.lang.MultiFn.invoke(MultiFn.java:234)
	at metabase.driver.sql_jdbc$fn__111186.invokeStatic(sql_jdbc.clj:76)
	at metabase.driver.sql_jdbc$fn__111186.invoke(sql_jdbc.clj:71)
	at clojure.lang.MultiFn.invoke(MultiFn.java:234)
	at metabase.sync.sync_metadata.sync_timezone$sync_timezone_BANG_.invokeStatic(sync_timezone.clj:41)
	at metabase.sync.sync_metadata.sync_timezone$sync_timezone_BANG_.invoke(sync_timezone.clj:36)
	at clojure.lang.AFn.applyToHelper(AFn.java:154)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:669)
	at clojure.core$apply.invoke(core.clj:662)
	at metabase.sync.util$run_step_with_metadata$fn__57032.doInvoke(util.clj:425)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:132)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:126)
	at metabase.sync.util$with_start_and_finish_debug_logging.invokeStatic(util.clj:150)
	at metabase.sync.util$with_start_and_finish_debug_logging.invoke(util.clj:146)
	at metabase.sync.util$run_step_with_metadata.invokeStatic(util.clj:420)
	at metabase.sync.util$run_step_with_metadata.invoke(util.clj:415)
	at metabase.sync.util$run_sync_operation$fn__57111.invoke(util.clj:531)
	at metabase.sync.util$run_sync_operation.invokeStatic(util.clj:529)
	at metabase.sync.util$run_sync_operation.invoke(util.clj:523)
	at metabase.sync.sync_metadata$sync_db_metadata_BANG_$fn__81408.invoke(sync_metadata.clj:63)
	at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:191)
	at metabase.sync.util$do_with_error_handling.invoke(util.clj:184)
	at clojure.core$partial$fn__5910.invoke(core.clj:2647)
	at metabase.driver$fn__48779.invokeStatic(driver.clj:725)
	at metabase.driver$fn__48779.invoke(driver.clj:725)
	at clojure.lang.MultiFn.invoke(MultiFn.java:239)
	at metabase.sync.util$sync_in_context$fn__56973.invoke(util.clj:167)
	at metabase.sync.util$with_db_logging_disabled$fn__56970.invoke(util.clj:159)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:132)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:126)
	at metabase.sync.util$with_start_and_finish_logging$fn__56957.invoke(util.clj:144)
	at metabase.sync.util$with_sync_events$fn__56952.invoke(util.clj:118)
	at metabase.sync.util$with_duplicate_ops_prevented$fn__56939.invoke(util.clj:90)
	at metabase.sync.util$do_sync_operation.invokeStatic(util.clj:216)
	at metabase.sync.util$do_sync_operation.invoke(util.clj:210)
	at metabase.sync.sync_metadata$sync_db_metadata_BANG_.invokeStatic(sync_metadata.clj:62)
	at metabase.sync.sync_metadata$sync_db_metadata_BANG_.invoke(sync_metadata.clj:58)
	at metabase.task.sync_databases$sync_and_analyze_database_STAR__BANG_.invokeStatic(sync_databases.clj:94)
	at metabase.task.sync_databases$sync_and_analyze_database_STAR__BANG_.invoke(sync_databases.clj:77)
	at metabase.task.sync_databases$sync_and_analyze_database_BANG_.invokeStatic(sync_databases.clj:113)
	at metabase.task.sync_databases$sync_and_analyze_database_BANG_.invoke(sync_databases.clj:101)
	at metabase.task.sync_databases.SyncAndAnalyzeDatabase.execute(sync_databases.clj:118)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
Caused by: org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException: Getting analyzing error from line 1, column 73 to line 1, column 173. Detail message: No matching function with signature: time_format(TIME, varchar).
	at org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException.of(MariaDbSqlException.java:34)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:195)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:178)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:322)
	at org.mariadb.jdbc.ClientSidePreparedStatement.executeInternal(ClientSidePreparedStatement.java:220)
	... 59 more
Caused by: java.sql.SQLException: Getting analyzing error from line 1, column 73 to line 1, column 173. Detail message: No matching function with signature: time_format(TIME, varchar).
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readErrorPacket(AbstractQueryProtocol.java:1693)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readPacket(AbstractQueryProtocol.java:1555)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.getResult(AbstractQueryProtocol.java:1518)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:319)
	... 60 more
```

Same issue on 
Starrocks : 3.2.2 
Metabase : 0.48.6

```
java.sql.SQLTransientConnectionException: (conn=3) Getting analyzing error from line 1, column 73 to line 1, column 173. Detail message: No matching function with signature: time_format(TIME, varchar).
        at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.createException(ExceptionFactory.java:79)
        at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.create(ExceptionFactory.java:158)
        at org.mariadb.jdbc.MariaDbStatement.executeExceptionEpilogue(MariaDbStatement.java:262)
        at org.mariadb.jdbc.ClientSidePreparedStatement.executeInternal(ClientSidePreparedStatement.java:229)
        at org.mariadb.jdbc.ClientSidePreparedStatement.execute(ClientSidePreparedStatement.java:149)
        at org.mariadb.jdbc.ClientSidePreparedStatement.executeQuery(ClientSidePreparedStatement.java:163)
        at com.mchange.v2.c3p0.impl.NewProxyPreparedStatement.executeQuery(NewProxyPreparedStatement.java:1471)
        at clojure.java.jdbc$execute_query_with_params.invokeStatic(jdbc.clj:1090)
        at clojure.java.jdbc$execute_query_with_params.invoke(jdbc.clj:1084)
        at clojure.java.jdbc$db_query_with_resultset_STAR_.invokeStatic(jdbc.clj:1113)
        at clojure.java.jdbc$db_query_with_resultset_STAR_.invoke(jdbc.clj:1093)
        at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1182)
        at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
        at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1160)
        at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
        at metabase.driver.mysql$fn__102784.invokeStatic(mysql.clj:205)
        at metabase.driver.mysql$fn__102784.invoke(mysql.clj:195)
        at clojure.lang.MultiFn.invoke(MultiFn.java:234)
        at metabase.driver.sql_jdbc$fn__105606.invokeStatic(sql_jdbc.clj:76)
        at metabase.driver.sql_jdbc$fn__105606.invoke(sql_jdbc.clj:71)
        at clojure.lang.MultiFn.invoke(MultiFn.java:234)
        at metabase.sync.sync_metadata.sync_timezone$sync_timezone_BANG_.invokeStatic(sync_timezone.clj:41)
        at metabase.sync.sync_metadata.sync_timezone$sync_timezone_BANG_.invoke(sync_timezone.clj:36)
        at clojure.lang.AFn.applyToHelper(AFn.java:154)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.core$apply.invokeStatic(core.clj:669)
        at clojure.core$apply.invoke(core.clj:662)
        at metabase.sync.util$run_step_with_metadata$fn__57009.doInvoke(util.clj:425)
        at clojure.lang.RestFn.invoke(RestFn.java:397)
        at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:132)
        at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:126)
        at metabase.sync.util$with_start_and_finish_debug_logging.invokeStatic(util.clj:150)
        at metabase.sync.util$with_start_and_finish_debug_logging.invoke(util.clj:146)
        at metabase.sync.util$run_step_with_metadata.invokeStatic(util.clj:420)
        at metabase.sync.util$run_step_with_metadata.invoke(util.clj:415)
        at metabase.sync.util$run_sync_operation$fn__57088.invoke(util.clj:531)
        at metabase.sync.util$run_sync_operation.invokeStatic(util.clj:529)
        at metabase.sync.util$run_sync_operation.invoke(util.clj:523)
        at metabase.sync.sync_metadata$sync_db_metadata_BANG_$fn__80792.invoke(sync_metadata.clj:63)
        at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:191)
        at metabase.sync.util$do_with_error_handling.invoke(util.clj:184)
        at clojure.core$partial$fn__5910.invoke(core.clj:2647)
        at metabase.driver$fn__48768.invokeStatic(driver.clj:725)
        at metabase.driver$fn__48768.invoke(driver.clj:725)
        at clojure.lang.MultiFn.invoke(MultiFn.java:239)
        at metabase.sync.util$sync_in_context$fn__56950.invoke(util.clj:167)
        at metabase.sync.util$with_db_logging_disabled$fn__56947.invoke(util.clj:159)
        at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:132)
        at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:126)
        at metabase.sync.util$with_start_and_finish_logging$fn__56934.invoke(util.clj:144)
        at metabase.sync.util$with_sync_events$fn__56929.invoke(util.clj:118)
        at metabase.sync.util$with_duplicate_ops_prevented$fn__56916.invoke(util.clj:90)
        at metabase.sync.util$do_sync_operation.invokeStatic(util.clj:216)
        at metabase.sync.util$do_sync_operation.invoke(util.clj:210)
        at metabase.sync.sync_metadata$sync_db_metadata_BANG_.invokeStatic(sync_metadata.clj:62)
        at metabase.sync.sync_metadata$sync_db_metadata_BANG_.invoke(sync_metadata.clj:58)
        at metabase.api.database$fn__97946$fn__97951.invoke(database.clj:998)
        at clojure.core$binding_conveyor_fn$fn__5823.invoke(core.clj:2047)
        at clojure.lang.AFn.call(AFn.java:18)
        at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
Caused by: org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException: Getting analyzing error from line 1, column 73 to line 1, column 173. Detail message: No matching function with signature: time_format(TIME, varchar).
        at org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException.of(MariaDbSqlException.java:34)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:195)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:178)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:322)
        at org.mariadb.jdbc.ClientSidePreparedStatement.executeInternal(ClientSidePreparedStatement.java:220)
        ... 59 more
Caused by: java.sql.SQLException: Getting analyzing error from line 1, column 73 to line 1, column 173. Detail message: No matching function with signature: time_format(TIME, varchar).
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readErrorPacket(AbstractQueryProtocol.java:1693)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readPacket(AbstractQueryProtocol.java:1555)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.getResult(AbstractQueryProtocol.java:1518)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:319)
        ... 60 more

```

@igorvpcleao 

I started to think that unsupported "time_format(TIME, varchar)" function may not the cause of this problem.

The reason that I think so is that I can see all tables in Starrocks 3.2.2 even it has same error output in the log below.

```
Caused by: org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException: Getting analyzing error from line 1, column 73 to line 1, column 173. Detail message: No matching function with signature: time_format(TIME, varchar).
        at org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException.of(MariaDbSqlException.java:34)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:195)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:178)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:322)
        at org.mariadb.jdbc.ClientSidePreparedStatement.executeInternal(ClientSidePreparedStatement.java:220)
        ... 59 more
Caused by: java.sql.SQLException: Getting analyzing error from line 1, column 73 to line 1, column 173. Detail message: No matching function with signature: time_format(TIME, varchar).
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readErrorPacket(AbstractQueryProtocol.java:1693)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readPacket(AbstractQueryProtocol.java:1555)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.getResult(AbstractQueryProtocol.java:1518)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:319)
        ... 60 more
```


The only change I found is version.
3.2.2 -> working.
3.2.3 -> not working.

In 3.2.3 , it doesn't look table scanning find anything in fingerprint-fields, classify-fields and classify-tables stage.

![image](https://github.com/StarRocks/starrocks/assets/16459517/6159bf93-f0ea-45c6-bb4c-2337598164ca)

But in 3.2.2, it successfully find tables and columns.

![image](https://github.com/StarRocks/starrocks/assets/16459517/4879fb57-ca45-4146-b17f-b39132fcde8f)

--- a/be/src/exprs/time_functions.cpp
+++ b/be/src/exprs/time_functions.cpp
@@ -3222,4 +3222,85 @@ Status TimeFunctions::last_day_close(FunctionContext* context, FunctionContext::
     return Status::OK();
 }
 
+// Format a time value according to a format string
+StatusOr<ColumnPtr> TimeFunctions::format_time(FunctionContext* context, const starrocks::Columns& columns) {
+    if (columns.size() != 2) {
+        return Status::InvalidArgument("FORMAT_TIME requires exactly 2 arguments");
+    }
+
+    const auto& time_column = columns[0];
+    const auto& format_column = columns[1];
+
+    if (time_column->is_nullable() || format_column->is_nullable()) {
+        return Status::InvalidArgument("FORMAT_TIME arguments cannot be NULL");
+    }
+
+    auto* time_col = ColumnHelper::cast_to<TYPE_TIME>(time_column);
+    auto* format_col = ColumnHelper::cast_to<TYPE_VARCHAR>(format_column);
+
+    const size_t size = time_column->size();
+    auto builder = ColumnHelper::get_builder<TYPE_VARCHAR>(size);
+
+    for (size_t i = 0; i < size; ++i) {
+        TimeValue time_val = time_col->get_time(i);
+        std::string_view format_str = format_col->get_slice(i);
+
+        // Convert TimeValue to hours, minutes, seconds
+        int hours = time_val.hour();
+        int minutes = time_val.minute();
+        int seconds = time_val.second();
+        int microseconds = time_val.microsecond();
+
+        std::stringstream result;
+        bool in_format = false;
+
+        for (size_t j = 0; j < format_str.size(); ++j) {
+            char c = format_str[j];
+            if (c == '%') {
+                in_format = true;
+                continue;
+            }
+
+            if (!in_format) {
+                result << c;
+                continue;
+            }
+
+            in_format = false;
+            switch (c) {
+                case 'H': // Hour (00-23)
+                    result << std::setfill('0') << std::setw(2) << hours;
+                    break;
+                case 'h': // Hour (01-12)
+                    result << std::setfill('0') << std::setw(2) << (((hours % 12) == 0) ? 12 : (hours % 12));
+                    break;
+                case 'i': // Minutes (00-59)
+                    result << std::setfill('0') << std::setw(2) << minutes;
+                    break;
+                case 'S': // Seconds (00-59)
+                case 's': // Seconds (00-59)
+                    result << std::setfill('0') << std::setw(2) << seconds;
+                    break;
+                case 'f': // Microseconds (000000-999999)
+                    result << std::setfill('0') << std::setw(6) << microseconds;
+                    break;
+                case 'p': // AM or PM
+                    result << (hours < 12 ? "AM" : "PM");
+                    break;
+                default:
+                    result << '%' << c;
+                    break;
+            }
+        }
+
+        if (in_format) {
+            result << '%'; // Handle trailing %
+        }
+
+        builder->append(result.str());
+    }
+
+    return builder->build();
+}
+
 } // namespace starrocks

--- a/be/src/exprs/time_functions.cpp
+++ b/be/src/exprs/time_functions.cpp
@@ -3237,22 +3237,27 @@ StatusOr<ColumnPtr> TimeFunctions::format_time(FunctionContext* context, const s
     auto format_viewer = ColumnViewer<TYPE_VARCHAR>(format_column);
 
     const size_t size = time_column->size();
-    auto builder = ColumnHelper::get_builder<TYPE_VARCHAR>(size);
+    auto builder = ColumnBuilder<TYPE_VARCHAR>(size);
 
     for (size_t i = 0; i < size; ++i) {
-        TimeValue time_val = time_viewer.value(i);
-        std::string_view format_str = format_viewer.value(i);
-
         if (time_viewer.is_null(i) || format_viewer.is_null(i)) {
-            builder->append_null();
+            builder.append_null();
             continue;
         }
 
+        TimestampValue time_val;
+        time_val.set_timestamp(time_viewer.value(i));
+        std::string_view format_str = format_viewer.value(i);
+
         // Convert TimeValue to hours, minutes, seconds
-        int hours = time_val.hour();
-        int minutes = time_val.minute();
-        int seconds = time_val.second();
-        int microseconds = time_val.microsecond();
+        int year;
+        int month;
+        int day;
+        int hours;
+        int minutes;
+        int seconds;
+        int microseconds;
+        time_val.to_timestamp(&year, &month, &day, &hours, &minutes, &seconds, &microseconds);
 
         std::stringstream result;
         bool in_format = false;
@@ -3300,10 +3305,10 @@ StatusOr<ColumnPtr> TimeFunctions::format_time(FunctionContext* context, const s
             result << '%'; // Handle trailing %
         }
 
-        builder->append(result.str());
+        builder.append(result.str());
     }
 
-    return builder->build();
+    return builder.build(ColumnHelper::is_all_const(columns));
 }
 
 } // namespace starrocks

--- a/be/src/exprs/time_functions.cpp
+++ b/be/src/exprs/time_functions.cpp
@@ -3223,7 +3223,7 @@ Status TimeFunctions::last_day_close(FunctionContext* context, FunctionContext::
 }
 
 // Format a time value according to a format string
-StatusOr<ColumnPtr> TimeFunctions::format_time(FunctionContext* context, const starrocks::Columns& columns) {
+StatusOr<ColumnPtr> TimeFunctions::time_format(FunctionContext* context, const starrocks::Columns& columns) {
     if (columns.size() != 2) {
         return Status::InvalidArgument("FORMAT_TIME requires exactly 2 arguments");
     }

--- a/be/src/exprs/time_functions.cpp
+++ b/be/src/exprs/time_functions.cpp
@@ -3231,9 +3231,7 @@ StatusOr<ColumnPtr> TimeFunctions::format_time(FunctionContext* context, const s
     const auto& time_column = columns[0];
     const auto& format_column = columns[1];
 
-    if (time_column->is_nullable() || format_column->is_nullable()) {
-        return Status::InvalidArgument("FORMAT_TIME arguments cannot be NULL");
-    }
+        RETURN_IF_COLUMNS_ONLY_NULL(columns);
 
     auto* time_col = ColumnHelper::cast_to<TYPE_TIME>(time_column);
     auto* format_col = ColumnHelper::cast_to<TYPE_VARCHAR>(format_column);

--- a/be/src/exprs/time_functions.cpp
+++ b/be/src/exprs/time_functions.cpp
@@ -3231,18 +3231,23 @@ StatusOr<ColumnPtr> TimeFunctions::format_time(FunctionContext* context, const s
     const auto& time_column = columns[0];
     const auto& format_column = columns[1];
 
-        RETURN_IF_COLUMNS_ONLY_NULL(columns);
-
-    auto* time_col = ColumnHelper::cast_to<TYPE_TIME>(time_column);
-    auto* format_col = ColumnHelper::cast_to<TYPE_VARCHAR>(format_column);
+    RETURN_IF_COLUMNS_ONLY_NULL(columns);
+    
+    auto time_viewer = ColumnViewer<TYPE_TIME>(time_column);
+    auto format_viewer = ColumnViewer<TYPE_VARCHAR>(format_column);
 
     const size_t size = time_column->size();
     auto builder = ColumnHelper::get_builder<TYPE_VARCHAR>(size);
 
     for (size_t i = 0; i < size; ++i) {
-        TimeValue time_val = time_col->get_time(i);
-        std::string_view format_str = format_col->get_slice(i);
+        TimeValue time_val = time_viewer.value(i);
+        std::string_view format_str = format_viewer.value(i);
 
+        if (time_viewer.is_null(i) || format_viewer.is_null(i)) {
+            builder->append_null();
+            continue;
+        }
+        
         // Convert TimeValue to hours, minutes, seconds
         int hours = time_val.hour();
         int minutes = time_val.minute();

--- a/be/src/exprs/time_functions.cpp
+++ b/be/src/exprs/time_functions.cpp
@@ -3232,7 +3232,7 @@ StatusOr<ColumnPtr> TimeFunctions::format_time(FunctionContext* context, const s
     const auto& format_column = columns[1];
 
     RETURN_IF_COLUMNS_ONLY_NULL(columns);
-    
+
     auto time_viewer = ColumnViewer<TYPE_TIME>(time_column);
     auto format_viewer = ColumnViewer<TYPE_VARCHAR>(format_column);
 
@@ -3247,7 +3247,7 @@ StatusOr<ColumnPtr> TimeFunctions::format_time(FunctionContext* context, const s
             builder->append_null();
             continue;
         }
-        
+
         // Convert TimeValue to hours, minutes, seconds
         int hours = time_val.hour();
         int minutes = time_val.minute();
@@ -3271,28 +3271,28 @@ StatusOr<ColumnPtr> TimeFunctions::format_time(FunctionContext* context, const s
 
             in_format = false;
             switch (c) {
-                case 'H': // Hour (00-23)
-                    result << std::setfill('0') << std::setw(2) << hours;
-                    break;
-                case 'h': // Hour (01-12)
-                    result << std::setfill('0') << std::setw(2) << (((hours % 12) == 0) ? 12 : (hours % 12));
-                    break;
-                case 'i': // Minutes (00-59)
-                    result << std::setfill('0') << std::setw(2) << minutes;
-                    break;
-                case 'S': // Seconds (00-59)
-                case 's': // Seconds (00-59)
-                    result << std::setfill('0') << std::setw(2) << seconds;
-                    break;
-                case 'f': // Microseconds (000000-999999)
-                    result << std::setfill('0') << std::setw(6) << microseconds;
-                    break;
-                case 'p': // AM or PM
-                    result << (hours < 12 ? "AM" : "PM");
-                    break;
-                default:
-                    result << '%' << c;
-                    break;
+            case 'H': // Hour (00-23)
+                result << std::setfill('0') << std::setw(2) << hours;
+                break;
+            case 'h': // Hour (01-12)
+                result << std::setfill('0') << std::setw(2) << (((hours % 12) == 0) ? 12 : (hours % 12));
+                break;
+            case 'i': // Minutes (00-59)
+                result << std::setfill('0') << std::setw(2) << minutes;
+                break;
+            case 'S': // Seconds (00-59)
+            case 's': // Seconds (00-59)
+                result << std::setfill('0') << std::setw(2) << seconds;
+                break;
+            case 'f': // Microseconds (000000-999999)
+                result << std::setfill('0') << std::setw(6) << microseconds;
+                break;
+            case 'p': // AM or PM
+                result << (hours < 12 ? "AM" : "PM");
+                break;
+            default:
+                result << '%' << c;
+                break;
             }
         }
 

--- a/be/src/exprs/time_functions.h
+++ b/be/src/exprs/time_functions.h
@@ -805,7 +805,7 @@ public:
      * @paramType columns: [TYPE_TIME, TYPE_VARCHAR]
      * @return ColumnPtr A column holding formatted time strings.
      */
-    DEFINE_VECTORIZED_FN(format_time);
+    DEFINE_VECTORIZED_FN(time_format);
 
 private:
     DEFINE_VECTORIZED_FN_TEMPLATE(_t_from_unix_to_datetime);

--- a/be/src/exprs/time_functions.h
+++ b/be/src/exprs/time_functions.h
@@ -799,6 +799,14 @@ public:
     // so this value is 253402329599(UTC 9999-12-31 23:59:59) - 24 * 3600(for all timezones)
     constexpr static const int64_t MAX_UNIX_TIMESTAMP = 253402243199L;
 
+    /**
+     * Format a time value according to a format string.
+     * @param: [time_value, format_str]
+     * @paramType columns: [TYPE_TIME, TYPE_VARCHAR]
+     * @return ColumnPtr A column holding formatted time strings.
+     */
+    DEFINE_VECTORIZED_FN(format_time);
+
 private:
     DEFINE_VECTORIZED_FN_TEMPLATE(_t_from_unix_to_datetime);
 

--- a/be/test/exprs/time_functions_test.cpp
+++ b/be/test/exprs/time_functions_test.cpp
@@ -3930,12 +3930,12 @@ TEST_F(TimeFunctionsTest, formatTimeTest) {
         auto time_builder = ColumnBuilder<TYPE_TIME>(1);
         TimestampValue ts = TimestampValue::create(0, 0, 0, 14, 30, 45);
         time_builder.append(ts.timestamp());
-        auto time_column = time_builder.build();
+        auto time_column = time_builder.build(false);
 
         // Create format column with basic format string
         auto format_builder = ColumnBuilder<TYPE_VARCHAR>(1);
         format_builder.append("%H:%i:%S");
-        auto format_column = format_builder.build();
+        auto format_column = format_builder.build(false);
 
         // Set up columns and function context
         Columns columns;
@@ -3961,7 +3961,7 @@ TEST_F(TimeFunctionsTest, formatTimeTest) {
         for (int i = 0; i < 4; i++) {
             time_builder.append(ts.timestamp());
         }
-        auto time_column = time_builder.build();
+        auto time_column = time_builder.build(false);
 
         // Create format column with different format strings
         auto format_builder = ColumnBuilder<TYPE_VARCHAR>(4);
@@ -3969,7 +3969,7 @@ TEST_F(TimeFunctionsTest, formatTimeTest) {
         format_builder.append("%H:%i");
         format_builder.append("Time: %H:%i");
         format_builder.append("%H");
-        auto format_column = format_builder.build();
+        auto format_column = format_builder.build(false);
 
         // Set up columns and function context
         Columns columns;

--- a/be/test/exprs/time_functions_test.cpp
+++ b/be/test/exprs/time_functions_test.cpp
@@ -25,6 +25,7 @@
 #include "column/binary_column.h"
 #include "column/column_builder.h"
 #include "column/column_helper.h"
+#include "column/column_viewer.h"
 #include "column/const_column.h"
 #include "column/fixed_length_column.h"
 #include "column/vectorized_fwd.h"
@@ -3928,7 +3929,7 @@ TEST_F(TimeFunctionsTest, formatTimeTest) {
     {
         // Create time column
         auto time_builder = ColumnBuilder<TYPE_TIME>(1);
-        TimestampValue ts = TimestampValue::create(0, 0, 0, 14, 30, 45);
+        TimestampValue ts = TimestampValue::create(0, 0, 0, 14, 30, 40);
         time_builder.append(ts.timestamp());
         auto time_column = time_builder.build(false);
 
@@ -3950,14 +3951,14 @@ TEST_F(TimeFunctionsTest, formatTimeTest) {
         // Verify result
         ASSERT_TRUE(result->is_binary());
         auto result_viewer = ColumnViewer<TYPE_VARCHAR>(result);
-        EXPECT_EQ("14:30:45", std::string(result_viewer.get_slice(0)));
+        EXPECT_EQ("14:30:40", std::string(result_viewer.value(0)));
     }
 
     // Multiple format strings test
     {
         // Create time column with multiple rows
         auto time_builder = ColumnBuilder<TYPE_TIME>(4);
-        TimestampValue ts = TimestampValue::create(0, 0, 0, 14, 30, 45);
+        TimestampValue ts = TimestampValue::create(0, 0, 0, 14, 30, 40);
         for (int i = 0; i < 4; i++) {
             time_builder.append(ts.timestamp());
         }
@@ -3984,10 +3985,10 @@ TEST_F(TimeFunctionsTest, formatTimeTest) {
         // Verify results
         ASSERT_TRUE(result->is_binary());
         auto result_viewer = ColumnViewer<TYPE_VARCHAR>(result);
-        EXPECT_EQ("14:30:45", std::string(result_viewer.get_slice(0)));
-        EXPECT_EQ("14:30", std::string(result_viewer.get_slice(1)));
-        EXPECT_EQ("Time: 14:30", std::string(result_viewer.get_slice(2)));
-        EXPECT_EQ("14", std::string(result_viewer.get_slice(3)));
+        EXPECT_EQ("14:30:40", std::string(result_viewer.value(0)));
+        EXPECT_EQ("14:30", std::string(result_viewer.value(1)));
+        EXPECT_EQ("Time: 14:30", std::string(result_viewer.value(2)));
+        EXPECT_EQ("14", std::string(result_viewer.value(3)));
     }
 }
 } // namespace starrocks

--- a/be/test/exprs/time_functions_test.cpp
+++ b/be/test/exprs/time_functions_test.cpp
@@ -3944,7 +3944,7 @@ TEST_F(TimeFunctionsTest, formatTimeTest) {
 
         // Execute format_time function
         TimeFunctions::format_prepare(_utils->get_fn_ctx(), FunctionContext::FunctionStateScope::FRAGMENT_LOCAL);
-        ColumnPtr result = TimeFunctions::format_time(_utils->get_fn_ctx(), columns).value();
+        ColumnPtr result = TimeFunctions::time_format(_utils->get_fn_ctx(), columns).value();
         TimeFunctions::format_close(_utils->get_fn_ctx(), FunctionContext::FunctionStateScope::FRAGMENT_LOCAL);
 
         // Verify result
@@ -3978,7 +3978,7 @@ TEST_F(TimeFunctionsTest, formatTimeTest) {
 
         // Execute format_time function
         TimeFunctions::format_prepare(_utils->get_fn_ctx(), FunctionContext::FunctionStateScope::FRAGMENT_LOCAL);
-        ColumnPtr result = TimeFunctions::format_time(_utils->get_fn_ctx(), columns).value();
+        ColumnPtr result = TimeFunctions::time_format(_utils->get_fn_ctx(), columns).value();
         TimeFunctions::format_close(_utils->get_fn_ctx(), FunctionContext::FunctionStateScope::FRAGMENT_LOCAL);
 
         // Verify results

--- a/be/test/exprs/time_functions_test.cpp
+++ b/be/test/exprs/time_functions_test.cpp
@@ -23,6 +23,7 @@
 #include <vector>
 
 #include "column/binary_column.h"
+#include "column/column_builder.h"
 #include "column/column_helper.h"
 #include "column/const_column.h"
 #include "column/fixed_length_column.h"
@@ -34,6 +35,7 @@
 #include "runtime/runtime_state.h"
 #include "runtime/time_types.h"
 #include "testutil/function_utils.h"
+#include "types/date_value.h"
 #include "types/logical_type.h"
 
 namespace starrocks {
@@ -3925,13 +3927,13 @@ TEST_F(TimeFunctionsTest, formatTimeTest) {
     // Basic format test
     {
         // Create time column
-        auto time_builder = ColumnHelper::get_builder<TYPE_TIME>(1);
-        TimeValue time_val(14, 30, 45, 0);
-        time_builder->append(time_val);
+        auto time_builder = ColumnBuilder<TYPE_TIME>(1);
+        TimestampValue ts = TimestampValue::create(0, 0, 0, 14, 30, 45);
+        time_builder->append(ts.timestamp());
         auto time_column = time_builder->build();
 
         // Create format column with basic format string
-        auto format_builder = ColumnHelper::get_builder<TYPE_VARCHAR>(1);
+        auto format_builder = ColumnBuilder<TYPE_VARCHAR>(1);
         format_builder->append("%H:%i:%S");
         auto format_column = format_builder->build();
 
@@ -3954,15 +3956,15 @@ TEST_F(TimeFunctionsTest, formatTimeTest) {
     // Multiple format strings test
     {
         // Create time column with multiple rows
-        auto time_builder = ColumnHelper::get_builder<TYPE_TIME>(4);
-        TimeValue time_val(14, 30, 45, 0);
+        auto time_builder = ColumnBuilder<TYPE_TIME>(4);
+        TimestampValue ts = TimestampValue::create(0, 0, 0, 14, 30, 45);
         for (int i = 0; i < 4; i++) {
-            time_builder->append(time_val);
+            time_builder->append(ts.timestamp());
         }
         auto time_column = time_builder->build();
 
         // Create format column with different format strings
-        auto format_builder = ColumnHelper::get_builder<TYPE_VARCHAR>(4);
+        auto format_builder = ColumnBuilder<TYPE_VARCHAR>(4);
         format_builder->append("%H:%i:%S");
         format_builder->append("%H:%i");
         format_builder->append("Time: %H:%i");

--- a/be/test/exprs/time_functions_test.cpp
+++ b/be/test/exprs/time_functions_test.cpp
@@ -3929,13 +3929,13 @@ TEST_F(TimeFunctionsTest, formatTimeTest) {
         // Create time column
         auto time_builder = ColumnBuilder<TYPE_TIME>(1);
         TimestampValue ts = TimestampValue::create(0, 0, 0, 14, 30, 45);
-        time_builder->append(ts.timestamp());
-        auto time_column = time_builder->build();
+        time_builder.append(ts.timestamp());
+        auto time_column = time_builder.build();
 
         // Create format column with basic format string
         auto format_builder = ColumnBuilder<TYPE_VARCHAR>(1);
-        format_builder->append("%H:%i:%S");
-        auto format_column = format_builder->build();
+        format_builder.append("%H:%i:%S");
+        auto format_column = format_builder.build();
 
         // Set up columns and function context
         Columns columns;
@@ -3959,17 +3959,17 @@ TEST_F(TimeFunctionsTest, formatTimeTest) {
         auto time_builder = ColumnBuilder<TYPE_TIME>(4);
         TimestampValue ts = TimestampValue::create(0, 0, 0, 14, 30, 45);
         for (int i = 0; i < 4; i++) {
-            time_builder->append(ts.timestamp());
+            time_builder.append(ts.timestamp());
         }
-        auto time_column = time_builder->build();
+        auto time_column = time_builder.build();
 
         // Create format column with different format strings
         auto format_builder = ColumnBuilder<TYPE_VARCHAR>(4);
-        format_builder->append("%H:%i:%S");
-        format_builder->append("%H:%i");
-        format_builder->append("Time: %H:%i");
-        format_builder->append("%H");
-        auto format_column = format_builder->build();
+        format_builder.append("%H:%i:%S");
+        format_builder.append("%H:%i");
+        format_builder.append("Time: %H:%i");
+        format_builder.append("%H");
+        auto format_column = format_builder.build();
 
         // Set up columns and function context
         Columns columns;

--- a/be/test/exprs/time_functions_test.cpp
+++ b/be/test/exprs/time_functions_test.cpp
@@ -3919,4 +3919,73 @@ TEST_F(TimeFunctionsTest, MakeDateTest) {
     ASSERT_TRUE(nullable_col->is_null(8));
     ASSERT_TRUE(nullable_col->is_null(9));
 }
+
+// Tests for format_time function
+TEST_F(TimeFunctionsTest, formatTimeTest) {
+    // Basic format test
+    {
+        // Create time column
+        auto time_builder = ColumnHelper::get_builder<TYPE_TIME>(1);
+        TimeValue time_val(14, 30, 45, 0);
+        time_builder->append(time_val);
+        auto time_column = time_builder->build();
+
+        // Create format column with basic format string
+        auto format_builder = ColumnHelper::get_builder<TYPE_VARCHAR>(1);
+        format_builder->append("%H:%i:%S");
+        auto format_column = format_builder->build();
+
+        // Set up columns and function context
+        Columns columns;
+        columns.emplace_back(time_column);
+        columns.emplace_back(format_column);
+
+        // Execute format_time function
+        TimeFunctions::format_prepare(_utils->get_fn_ctx(), FunctionContext::FunctionStateScope::FRAGMENT_LOCAL);
+        ColumnPtr result = TimeFunctions::format_time(_utils->get_fn_ctx(), columns).value();
+        TimeFunctions::format_close(_utils->get_fn_ctx(), FunctionContext::FunctionStateScope::FRAGMENT_LOCAL);
+
+        // Verify result
+        ASSERT_TRUE(result->is_binary());
+        auto result_viewer = ColumnViewer<TYPE_VARCHAR>(result);
+        EXPECT_EQ("14:30:45", std::string(result_viewer.get_slice(0)));
+    }
+
+    // Multiple format strings test
+    {
+        // Create time column with multiple rows
+        auto time_builder = ColumnHelper::get_builder<TYPE_TIME>(4);
+        TimeValue time_val(14, 30, 45, 0);
+        for (int i = 0; i < 4; i++) {
+            time_builder->append(time_val);
+        }
+        auto time_column = time_builder->build();
+
+        // Create format column with different format strings
+        auto format_builder = ColumnHelper::get_builder<TYPE_VARCHAR>(4);
+        format_builder->append("%H:%i:%S");
+        format_builder->append("%H:%i");
+        format_builder->append("Time: %H:%i");
+        format_builder->append("%H");
+        auto format_column = format_builder->build();
+
+        // Set up columns and function context
+        Columns columns;
+        columns.emplace_back(time_column);
+        columns.emplace_back(format_column);
+
+        // Execute format_time function
+        TimeFunctions::format_prepare(_utils->get_fn_ctx(), FunctionContext::FunctionStateScope::FRAGMENT_LOCAL);
+        ColumnPtr result = TimeFunctions::format_time(_utils->get_fn_ctx(), columns).value();
+        TimeFunctions::format_close(_utils->get_fn_ctx(), FunctionContext::FunctionStateScope::FRAGMENT_LOCAL);
+
+        // Verify results
+        ASSERT_TRUE(result->is_binary());
+        auto result_viewer = ColumnViewer<TYPE_VARCHAR>(result);
+        EXPECT_EQ("14:30:45", std::string(result_viewer.get_slice(0)));
+        EXPECT_EQ("14:30", std::string(result_viewer.get_slice(1)));
+        EXPECT_EQ("Time: 14:30", std::string(result_viewer.get_slice(2)));
+        EXPECT_EQ("14", std::string(result_viewer.get_slice(3)));
+    }
+}
 } // namespace starrocks

--- a/gensrc/script/functions.py
+++ b/gensrc/script/functions.py
@@ -559,6 +559,7 @@ vectorized_functions = [
     [50403, 'last_day', True, False, 'DATE', ['DATETIME', 'VARCHAR'], 'TimeFunctions::last_day_with_format',
      'TimeFunctions::last_day_prepare', 'TimeFunctions::last_day_close'],
     [50501, 'makedate', True, False, 'DATE', ['INT', 'INT'], 'TimeFunctions::make_date'],
+    [50610, 'makedate', True, False, 'VARCHAR', ['TIME', 'VARCHAR'], 'TimeFunctions::format_time'],
 
     # 60xxx: like predicate
     # important ref: LikePredicate.java, must keep name equals LikePredicate.Operator

--- a/gensrc/script/functions.py
+++ b/gensrc/script/functions.py
@@ -559,7 +559,7 @@ vectorized_functions = [
     [50403, 'last_day', True, False, 'DATE', ['DATETIME', 'VARCHAR'], 'TimeFunctions::last_day_with_format',
      'TimeFunctions::last_day_prepare', 'TimeFunctions::last_day_close'],
     [50501, 'makedate', True, False, 'DATE', ['INT', 'INT'], 'TimeFunctions::make_date'],
-    [50610, 'format_time', True, False, 'VARCHAR', ['TIME', 'VARCHAR'], 'TimeFunctions::format_time'],
+    [50610, 'time_format', True, False, 'VARCHAR', ['TIME', 'VARCHAR'], 'TimeFunctions::time_format'],
 
     # 60xxx: like predicate
     # important ref: LikePredicate.java, must keep name equals LikePredicate.Operator

--- a/gensrc/script/functions.py
+++ b/gensrc/script/functions.py
@@ -559,7 +559,7 @@ vectorized_functions = [
     [50403, 'last_day', True, False, 'DATE', ['DATETIME', 'VARCHAR'], 'TimeFunctions::last_day_with_format',
      'TimeFunctions::last_day_prepare', 'TimeFunctions::last_day_close'],
     [50501, 'makedate', True, False, 'DATE', ['INT', 'INT'], 'TimeFunctions::make_date'],
-    [50610, 'makedate', True, False, 'VARCHAR', ['TIME', 'VARCHAR'], 'TimeFunctions::format_time'],
+    [50610, 'format_time', True, False, 'VARCHAR', ['TIME', 'VARCHAR'], 'TimeFunctions::format_time'],
 
     # 60xxx: like predicate
     # important ref: LikePredicate.java, must keep name equals LikePredicate.Operator

